### PR TITLE
chore(ui-react): type withAuthenticator

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add missing typing to `withAuthenticator`

Screen Shot 2022-04-19 at 3 28 14 PM

#### Issue #, if available
NA

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by ErikCH — APPROVED** [review]
👍

**Comment by wlee221** [comment]
Yep +1 on the changeset, bc it changes the npm package content 🙏

File: packages/react/src/components/Authenticator/withAuthenticator.tsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import { Authenticator, AuthenticatorProps } from './Authenticator';
 
 export type WithAuthenticatorOptions = Omit<AuthenticatorProps, 'children'>;
 
-export function withAuthenticator(
-  Component,
+export function withAuthenticator<T>(
```
Comment (reesscot) [inline packages/react/src/components/Authenticator/withAuthenticator.tsx:5]: Can we use something other than `T` here? I'm not a fan of single letter generic type props.

Comment (reesscot) [inline packages/react/src/components/Authenticator/withAuthenticator.tsx:5]: Can we use something other than `T` here? I'm not a fan of single letter generic type params. Maybe something like:

```suggestion
export function withAuthenticator<ComponentProps>(
```

**Review by reesscot — APPROVED** [review]
LGTM

**Comment by drummerwolli** [comment]
Hi @calebpollman this PR introduces some typescript warnings and the examples dont explain how to fix them. This [SO issue](https://stackoverflow.com/questions/73631333/what-types-need-to-be-used-for-withauthenticator-in-a-typescript-amplify-react) from someone else is describing the same issue I have in my project. can you check the SO issue or provide guidance here in the PR on how to address this and remove the typescript warnings?

**Comment by reesscot** [comment]
@drummerwolli Those issues should be fixed by this PR: https://github.com/aws-amplify/amplify-ui/pull/2570 and is scheduled to be released this week. Would you mind testing the `next` tag and let us know if it fixes your issues?

```
npm install @aws-amplify/ui-react@next
```

**Comment by drummerwolli** [comment]
@reesscot unfortunately i still get the same typescript warnings as in the SO thread with the @next version :(

**Comment by calebpollman** [comment]
Hi @drummerwolli! What is the `@aws-amplify/ui-react` version in your _node_modules_? Was unable to reproduce the warnings from the SO thread using `@aws-amplify/ui-react@0.0.0-next-8231448-20220912013143`

**Comment by drummerwolli** [comment]
yup, i have the same:

Untitled

(`ProtectedApp` is `App`, just a different name)

TS version is 4.8.3, React is 18.2.0 ... in case those are interesting. let me know what other info could help to debug this.

**Comment by calebpollman** [comment]
@drummerwolli Updated the `withAuthenticator` typing [here](https://github.com/aws-amplify/amplify-ui/pull/2586), which is now available in `@aws-amplify/ui-react@next`. Can you install and see if it clears your issue?

**Comment by drummerwolli** [comment]
@calebpollman TS2345 is now fixed, thx! 🥳 

TS2739 still wants to set `signOut` and `user` props on the `App` object in `index.tsx`. I can "fix" this with `<App signOut={undefined} user={undefined} />`, but is this the intended way to handle this? Or do we have a more user-friendly way here?

**Comment by calebpollman** [comment]
@drummerwolli Glad to hear the initial issue is resolved. Regarding what you are seeing with  the `signOut`/`user` props, can you please provide me a minimal sample app that reproduces the issue? Having a hard time reproducing locally

**Pull request merged**